lattices/src/collections.rs

```diff
@@ -10,6 +11,47 @@ use cc_traits::{
     KeyedRef, Len, MapIter, MapIterMut, SimpleKeyedRef,
 };
 
+/// Trait for transforming the values of a map without changing the overal type of the data structure.
```

Comment:
nit: typo "overal"

Reply:
For all the below, are there cases where in-place mutation would be faster and more desirable than map? Should we support that in addition?